include/oneapi/tbb/detail/_concurrent_skip_list.h

```diff
@@ -72,25 +72,51 @@ class skip_list_node {
     using pointer = typename value_allocator_traits::pointer;
     using const_pointer = typename value_allocator_traits::const_pointer;
 
-    skip_list_node( size_type levels, container_allocator_type& alloc )
-        : my_container_allocator(alloc), my_height(levels), my_index_number(0)
-    {
-        for (size_type l = 0; l < my_height; ++l) {
-            allocator_traits::construct(my_container_allocator, &get_atomic_next(l), nullptr);
-        }
-    }
+    //In perfect world these constructor and destructor would have been private, 
+    //however this seems technically impractical due to use of allocator_traits.
 
-    ~skip_list_node() {
-        for (size_type l = 0; l < my_height; ++l) {
-            allocator_traits::destroy(my_container_allocator, &get_atomic_next(l));
-        }
-    }
+    //Should not be called directly, instead use create method
+    skip_list_node( size_type levels )
+        : my_height(levels), my_index_number(0)
+    {}
+
+    //Should not be called directly, instead use destroy method
+    ~skip_list_node() {}
 
     skip_list_node( const skip_list_node& ) = delete;
     skip_list_node( skip_list_node&& ) = delete;
     skip_list_node& operator=( const skip_list_node& ) = delete;
     skip_list_node& operator=( skip_list_node&& ) = delete;
 
+    static skip_list_node* create( container_allocator_type& alloc, size_type height ) {
+        size_type sz = calc_node_size(height);
+        auto* node = reinterpret_cast<skip_list_node*>(allocator_traits::allocate(alloc, sz));
```

Comment:
One more comment regarding this place: I would add `static_assert` that either `std::is_same_v<allocator_traits::value_type, std::uint8_t>` or `sizeof(allocator_traits::value_type) == sizeof(std::uint8_t)`.

Reply:
added